plugins/inputs/icecast/icecast.go

```diff
@@ -0,0 +1,197 @@
+package icecast
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// SourceListmounts holds single listener elements from the Icecast XML
+type SourceListmounts struct {
+	Mount       string `xml:"mount,attr"`
+	Listeners   int32  `xml:"listeners"`
+	Connected   int32  `xml:"connected"`
+	ContentType string `xml:"content-type"`
+}
+
+// Listmounts main structure of the icecast XML
+type Listmounts struct {
```

Comment:
If you really need to export this type please give it a more specific name (e.g. IcecastListmounts) to avoid collisions with other plugins.